source/common/router/config_impl.cc

```diff
@@ -864,6 +845,7 @@ RegexRouteEntryImpl::RegexRouteEntryImpl(
     Server::Configuration::ServerFactoryContext& factory_context,
     ProtobufMessage::ValidationVisitor& validator)
     : RouteEntryImplBase(vhost, route, factory_context, validator) {
+  // TODO(yangminzhu): Use PathMatcher once hidden_envoy_deprecated_regex is removed.
```

Comment:
Could you open an issue for that? Thanks!

Reply:
Done: https://github.com/envoyproxy/envoy/issues/10032